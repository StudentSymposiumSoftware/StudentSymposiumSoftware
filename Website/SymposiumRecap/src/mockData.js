/*

  Mock Data has many objects of these fields:
   - Author
   - Title
   - Category
   - Major
   - Professor
   - University
   - Abstract Number

*/


const mock_data = [{ author: 'Elijah Smith',
    professor: 'Olivia Lewis',
    school: 'UMaine Machias',
    major: 'Physics',
    title: 'Machine Learning Approaches to Financial Fraud Detection',
    category: 'Social Sciences and Humanities',
    abstractNumber: 38893888 },
  { author: 'Sophia Campbell',
    professor: 'Mia Lee',
    school: 'UMaine Machias',
    major: 'Political Science',
    title: 'The Psychology of Decision-Making in Uncertain Environments',
    category: 'Social Sciences and Humanities',
    abstractNumber: 75957852 },
  { author: 'Lucy Martinez',
    professor: 'Carter Martinez',
    school: 'UMaine Machias',
    major: 'Business Administration',
    title: 'The Sociology of Online Communities in the Digital Age',
    category: 'Natural Sciences',
    abstractNumber: 48203955 },
  { author: 'Zoe Jackson',
    professor: 'Layla Perez',
    school: 'UMaine Machias',
    major: 'Psychology',
    title: 'Advancements in Human-Robot Interaction for Assistive Technologies',
    category: 'Physical and Mathematical Sciences',
    abstractNumber: 64568884 },
  { author: 'Noah King',
    professor: 'William Smith',
    school: 'UMaine Machias',
    major: 'Political Science',
    title: 'Blockchain Technology in Supply Chain Transparency',
    category: 'Biomedical Sciences',
    abstractNumber: 40679302 },
  { author: 'Joseph Baker',
    professor: 'Aria Thompson',
    school: 'UMaine Orono',
    major: 'Mathematics',
    title: 'Advancements in Human-Robot Interaction for Assistive Technologies',
    category: 'Interdisciplinary Research (research across 3 or more different departments or colleges)',
    abstractNumber: 37625940 },
  { author: 'Ava Rodriguez',
    professor: 'Victoria Thomas',
    school: 'UMaine Machias',
    major: 'Physics',
    title: 'Climate Change and Its Effects on Global Food Security',
    category: 'Physical and Mathematical Sciences',
    abstractNumber: 35716422 },
  { author: 'Henry Adams',
    professor: 'Lucy Miller',
    school: 'UMaine Machias',
    major: 'Psychology',
    title: 'The Psychology of Decision-Making in Uncertain Environments',
    category: 'Business',
    abstractNumber: 82413071 },
  { author: 'Mason Hernandez',
    professor: 'Aiden Perez',
    school: 'UMaine Orono',
    major: 'Business Administration',
    title: 'Machine Learning Approaches to Financial Fraud Detection',
    category: 'Arts',
    abstractNumber: 27882383 },
  { author: 'Emma Taylor',
    professor: 'Ethan Hernandez',
    school: 'UMaine Orono',
    major: 'Biology',
    title: 'Machine Learning Approaches to Financial Fraud Detection',
    category: 'Business',
    abstractNumber: 92969314 },
  { author: 'Aurora Martinez',
    professor: 'William Anderson',
    school: 'UMaine Machias',
    major: 'Computer Science',
    title: 'The Role of Citizen Science in Biodiversity Conservation',
    category: 'Business',
    abstractNumber: 29718536 },
  { author: 'Lucas Robinson',
    professor: 'Scarlett King',
    school: 'UMaine Orono',
    major: 'Psychology',
    title: 'The Role of Citizen Science in Biodiversity Conservation',
    category: 'Natural Sciences',
    abstractNumber: 1457768 },
  { author: 'Chloe Adams',
    professor: 'Liam Rodriguez',
    school: 'UMaine Orono',
    major: 'Physics',
    title: 'Blockchain Technology in Supply Chain Transparency',
    category: 'Business',
    abstractNumber: 14148682 },
  { author: 'Layla Young',
    professor: 'Alexander Roberts',
    school: 'UMaine Orono',
    major: 'Biology',
    title: 'Exploring the Feasibility of Space-Based Solar Power',
    category: 'Engineering and Information Sciences (including Biomedical Engineering)',
    abstractNumber: 60454722 },
  { author: 'Alexander Hall',
    professor: 'Aria Clark',
    school: 'UMaine Machias',
    major: 'Business Administration',
    title: 'Social Media Influence on Adolescent Mental Health',
    category: 'Education',
    abstractNumber: 41498009 },
  { author: 'Zoe Young',
    professor: 'Amelia Jackson',
    school: 'UMaine Orono',
    major: 'Physics',
    title: 'Analyzing the Impact of Microplastics on Marine Ecosystems',
    category: 'Engineering and Information Sciences (including Biomedical Engineering)',
    abstractNumber: 23333306 },
  { author: 'Mason Clark',
    professor: 'Scarlett Hernandez',
    school: 'UMaine Machias',
    major: 'Nursing',
    title: 'Cultural Impacts on Language Evolution',
    category: 'Physical and Mathematical Sciences',
    abstractNumber: 97656877 },
  { author: 'Henry Hall',
    professor: 'Ella Garcia',
    school: 'UMaine Machias',
    major: 'Mechanical Engineering',
    title: 'Social Media Influence on Adolescent Mental Health',
    category: 'Interdisciplinary Research (research across 3 or more different departments or colleges)',
    abstractNumber: 91417686 },
  { author: 'Samuel Scott',
    professor: 'Evelyn Green',
    school: 'UMaine Machias',
    major: 'Mathematics',
    title: 'Advancements in Human-Robot Interaction for Assistive Technologies',
    category: 'Allied Health',
    abstractNumber: 4887031 },
  { author: 'Grace Nelson',
    professor: 'Aria Scott',
    school: 'UMaine Machias',
    major: 'Nursing',
    title: 'Advancements in Human-Robot Interaction for Assistive Technologies',
    category: 'Physical and Mathematical Sciences',
    abstractNumber: 36475226 },
  { author: 'Scarlett Hernandez',
    professor: 'Matthew Wright',
    school: 'UMaine Machias',
    major: 'Mechanical Engineering',
    title: 'Machine Learning Approaches to Financial Fraud Detection',
    category: 'Arts',
    abstractNumber: 29014475 },
  { author: 'Maya Hernandez',
    professor: 'William Green',
    school: 'UMaine Machias',
    major: 'Physics',
    title: 'The Future of CRISPR in Gene Therapy',
    category: 'Physical and Mathematical Sciences',
    abstractNumber: 44667664 },
  { author: 'Elijah Rivera',
    professor: 'Harper Martinez',
    school: 'UMaine Orono',
    major: 'Computer Science',
    title: 'Analyzing the Impact of Microplastics on Marine Ecosystems',
    category: 'Social Sciences and Humanities',
    abstractNumber: 21709418 },
  { author: 'Liam Taylor',
    professor: 'Ella Walker',
    school: 'UMaine Machias',
    major: 'Political Science',
    title: 'Social Media Influence on Adolescent Mental Health',
    category: 'Physical and Mathematical Sciences',
    abstractNumber: 45822327 },
  { author: 'Mia Robinson',
    professor: 'Noah Green',
    school: 'UMaine Orono',
    major: 'Mathematics',
    title: 'The Impact of Remote Work on Organizational Productivity',
    category: 'Physical and Mathematical Sciences',
    abstractNumber: 75710656 },
  { author: 'Evelyn Rodriguez',
    professor: 'Lucas Moore',
    school: 'UMaine Orono',
    major: 'Physics',
    title: 'Deep Learning for Medical Image Analysis',
    category: 'Allied Health',
    abstractNumber: 39877467 },
  { author: 'Zoe Brown',
    professor: 'Scarlett Thompson',
    school: 'UMaine Orono',
    major: 'Physics',
    title: 'Cultural Impacts on Language Evolution',
    category: 'Biomedical Sciences',
    abstractNumber: 74596190 },
  { author: 'Sophie Thomas',
    professor: 'David Rodriguez',
    school: 'UMaine Machias',
    major: 'Mechanical Engineering',
    title: 'Advancements in Human-Robot Interaction for Assistive Technologies',
    category: 'Physical and Mathematical Sciences',
    abstractNumber: 30556122 },
  { author: 'Maya Mitchell',
    professor: 'Aiden White',
    school: 'UMaine Orono',
    major: 'Nursing',
    title: 'Advancements in Human-Robot Interaction for Assistive Technologies',
    category: 'Arts',
    abstractNumber: 75180428 },
  { author: 'Jack Anderson',
    professor: 'Harper Rodriguez',
    school: 'UMaine Orono',
    major: 'Physics',
    title: 'Analyzing the Impact of Microplastics on Marine Ecosystems',
    category: 'Natural Sciences',
    abstractNumber: 82398661 }]

export const getMockData = () => {
    return mock_data;
}