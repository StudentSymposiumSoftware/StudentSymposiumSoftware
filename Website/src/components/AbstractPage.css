#author-abstract-page{
    color: white;
    text-decoration: none;
    cursor: auto;
}
