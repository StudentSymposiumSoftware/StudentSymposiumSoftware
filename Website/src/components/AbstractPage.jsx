import "./AbstractPage.css"
import { useNavigate, useParams  } from "react-router-dom"

function AbstractPage(props) {
    const navigate = useNavigate();

    let { id} = useParams();
    for (let i = 0; i < props.data.length; i++) {
        if (props.data[i].abstractNumber == id) {
        return (
            <div id="abstract-container">
                <h1 style={{ fontSize: "50px" }}>{props.data[i].title}</h1>
                <h2  className="author-link" id="author-abstract-page"><i>Authors:</i> {[props.data[i].author, props.data[i]["1st Co-author's Full Name"], props.data[i]["2nd Co-author's Full Name"], props.data[i]["3rd Co-author's Full Name"], props.data[i]["4th Co-author's Full Name"], props.data[i]["5th Co-author's Full Name"], props.data[i]["If you have more than 5 co-authors, please list their names and emails below, and whether they are an undergraduate student, graduate student, faculty, or external author. Please see the example."]].join(', ')}</h2>
                <h2><i>Faculty Advisor:</i> {props.data[i]["Faculty Mentor Name"]}</h2>
                <h2><i>Abstract Number:</i> {props.data[i].abstractNumber}</h2>
                <br></br>
                <p>{props.data[i]['Project Abstract']}</p>
            </div>
      );
    }
  }

  return (
    <div>
      <h1 style={{ fontSize: "50px" }}>
          Uh-oh! 
      </h1>
      <div style={{ fontSize: "50px" }}>
          Abstract {id} does not exist.
      </div>
    </div>
  );

}

export default AbstractPage