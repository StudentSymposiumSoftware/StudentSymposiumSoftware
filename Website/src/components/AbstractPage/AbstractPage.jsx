import "./AbstractPage.css";
import { useParams, useNavigate } from "react-router-dom";

import { findSchoolLogo } from "../shared";

function AbstractPage({data}) {
    const navigate = useNavigate();
    var abstractNumber = useParams().number;
    data = data.filter(item => {
        return item.abstractNumber == abstractNumber;
    });
    if (data.length != 0) {
        data = data[0];
    }

    return (
        <div className="bg-white rounded-lg mt-[0px] m-5 h-[calc(100vh-160px)] text-black p-3">
            {data.length != 0 && (
                <div className="grid grid-cols-3">
                    <div className="col-span-2">
                        <a className="cursor-pointer all-link" onClick={() => navigate(-1)}>Return to Previous Page</a><br/>
                        <span className="text-[24px]">{data.major} {abstractNumber} - {data.title}</span><br/>
                        <span className="text-[20px] italic">{data.allAuthors}</span><br/>
                        <span className="text-[20px] italic">Mentor: {data.professor}</span>
                    </div>
                    <img src={findSchoolLogo(data.school)} className="place-self-center"/>
                    <div className="col-span-3">
                        <span className="block mb-3 mt-2">{data.keywords.map((keyword) => <Keyword keyword={keyword.trim()} key={keyword}/>)}</span>
                        {data.abstract}
                    </div>
                </div>
            )}
        </div>
    )
}

function Keyword({keyword}) {
    return (
        <span className="bg-primary-background text-white rounded-lg p-[5px] m-[5px] mb-0 ml-0">
            {keyword}
        </span>
    )
}

export default AbstractPage