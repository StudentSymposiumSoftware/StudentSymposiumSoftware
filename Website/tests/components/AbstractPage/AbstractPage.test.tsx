import React from 'react';
import { render } from "@testing-library/react";
import { expect, it, vi } from "vitest";

import mockData from '../../../src/mockData';
import AbstractPage from '../../../src/components/AbstractPage/AbstractPage';

vi.mock('react-router-dom', async () => {
    const mod = await vi.importActual('react-router-dom');
    return {
      ...mod,
      useParams: () => ({
        number: 38893888,
      }),
      useNavigate: vi.fn(),
    };
  });

it('Abstract Page renders correctly', () => {
    const tree = render(<AbstractPage data={mockData}/>)
    expect(tree).toMatchSnapshot();
});