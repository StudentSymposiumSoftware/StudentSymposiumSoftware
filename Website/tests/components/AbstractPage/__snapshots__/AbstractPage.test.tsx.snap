// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Abstract Page renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="bg-white rounded-lg mt-[0px] m-5 h-[calc(100vh-160px)] text-black p-3"
      >
        <div
          class="grid grid-cols-3"
        >
          <div
            class="col-span-2"
          >
            <a
              class="cursor-pointer all-link"
            >
              Return to Previous Page
            </a>
            <br />
            <span
              class="text-[24px]"
            >
              Physics
               
              38893888
               - 
              Machine Learning Approaches to Financial Fraud Detection
            </span>
            <br />
            <span
              class="text-[20px] italic"
            >
              Elijah Smith
              Olivia Lewis
            </span>
            <br />
            <span
              class="text-[20px] italic"
            >
              Mentor: 
              Olivia Lewis
            </span>
          </div>
          <img
            class="place-self-center"
            src="/StudentSymposiumSoftware/umaine-machias.png"
          />
          <div
            class="col-span-3"
          >
            <span
              class="block mb-3 mt-2"
            >
              <span
                class="bg-primary-background text-white rounded-lg p-[5px] m-[5px] mb-0 ml-0"
              >
                Machine Learning
              </span>
              <span
                class="bg-primary-background text-white rounded-lg p-[5px] m-[5px] mb-0 ml-0"
              >
                Financial Fraud
              </span>
              <span
                class="bg-primary-background text-white rounded-lg p-[5px] m-[5px] mb-0 ml-0"
              >
                Detection
              </span>
            </span>
            This research explores the application of machine learning techniques in detecting financial fraud. By analyzing large datasets, we developed algorithms that can identify patterns indicative of fraudulent activities. Our findings demonstrate the effectiveness of these approaches in enhancing fraud detection systems, potentially saving millions in losses.
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="bg-white rounded-lg mt-[0px] m-5 h-[calc(100vh-160px)] text-black p-3"
    >
      <div
        class="grid grid-cols-3"
      >
        <div
          class="col-span-2"
        >
          <a
            class="cursor-pointer all-link"
          >
            Return to Previous Page
          </a>
          <br />
          <span
            class="text-[24px]"
          >
            Physics
             
            38893888
             - 
            Machine Learning Approaches to Financial Fraud Detection
          </span>
          <br />
          <span
            class="text-[20px] italic"
          >
            Elijah Smith
            Olivia Lewis
          </span>
          <br />
          <span
            class="text-[20px] italic"
          >
            Mentor: 
            Olivia Lewis
          </span>
        </div>
        <img
          class="place-self-center"
          src="/StudentSymposiumSoftware/umaine-machias.png"
        />
        <div
          class="col-span-3"
        >
          <span
            class="block mb-3 mt-2"
          >
            <span
              class="bg-primary-background text-white rounded-lg p-[5px] m-[5px] mb-0 ml-0"
            >
              Machine Learning
            </span>
            <span
              class="bg-primary-background text-white rounded-lg p-[5px] m-[5px] mb-0 ml-0"
            >
              Financial Fraud
            </span>
            <span
              class="bg-primary-background text-white rounded-lg p-[5px] m-[5px] mb-0 ml-0"
            >
              Detection
            </span>
          </span>
          This research explores the application of machine learning techniques in detecting financial fraud. By analyzing large datasets, we developed algorithms that can identify patterns indicative of fraudulent activities. Our findings demonstrate the effectiveness of these approaches in enhancing fraud detection systems, potentially saving millions in losses.
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
